### PR TITLE
Fix DESIGN.md frontmatter coverage

### LINKED ISSUE #436: doctor.mjs and the detector disagree on where design values live — body vs frontmatter
**Plugin version:** impeccable 4.0.2

## Summary

`doctor.mjs` and the detector disagree about where a project's design system is documented. The
doctor reads the `DESIGN.md` markdown **body**; the detector's `design-system-*` rules read the
YAML **frontmatter**. A project can therefore be fully documented as far as the detector is
concerned and simultaneously reported as undocumented by the doctor, with no way to satisfy both
without duplicating design values.

## The two readers

**Doctor.** `design-md-coverage` comes from `checkDesignCoverage` in
`skills/impeccable/scripts/lib/staleness-deep.mjs` (~:126-130):

```js
const missing = ['colors', 'typography', 'components']
  .filter((section) => !model[section]);
```

`model` is built by `parseDesignMd` (`lib/design-parser.mjs:826`), which splits the **body** into
sections and calls `extractColors(sections['Colors'])`, `extractTypography(sections['Typography'])`,
`extractComponents(sections['Components'])`. Frontmatter is parsed into `model.frontmatter` and is
never consulted by this check.

**Detector.** The token-conformance rules (`design-system-color`, `design-system-radius`) arm off
the frontmatter.

## Differential evidence

Measured on a real project:

- **Sidecar (`.impeccable/design.json`) present, `DESIGN.md` body a stub, no frontmatter:** a
  planted `#FF00FF` and a planted `23px` produced **0** findings from the token-conformance rules.
- **Frontmatter present:** **exactly those 2** planted violations flagged, and the project's canon
  values clean — no false positives.

The same project's `doctor.mjs --json` reports `design-md-coverage` for all three sections
throughout, because the body carries no `## Colors` / `## Typography` / `## Components` headings.

## Why this shape is legitimate

A derived-mirror `DESIGN.md` — frontmatter carrying the canon, a short pointer body, and a single
upstream source of truth for the values — is a reasonable design for a project whose design
authority lives elsewhere (in our case a reviewed rules document, with a committed test locking
three-way parity across it, the frontmatter, and the sidecar).

That shape satisfies the detector and permanently fails the doctor. Worse, the doctor's own
suggested fix — "`document` fills it from the code if the project has the answer in its CSS" —
would write design values into a second location, which is exactly what such a project is
structurally avoiding. Following the advice would introduce the defect the design prevents.

## Secondary observation: the check is a heading scan

In practice `design-md-coverage` tests for **headings**, not for documentation. Each extractor has
exactly two exits — `return null` when the section is absent (`lib/design-parser.mjs:358`, `:519`,
`:696`) and an unconditional object literal otherwise (`:405`, `:570`, `:733`) — so a section that
exists and parses to **zero** values is still truthy and silences the finding.

An empty `## Colors` heading with no content therefore satisfies the check. That is the workaround
we adopted (a heading plus a one-line pointer, no values), and it works — but it means the finding
measures heading presence rather than whether guidance exists, which is presumably not the intent.

## Suggested fix

1. Have `checkDesignCoverage` treat frontmatter coverage as satisfying, so the two readers agree.
2. Failing that, state in the finding text that frontmatter is not counted, so the intent is at
   least legible to someone deciding how to respond.
3. Consider whether a values-free heading should satisfy the check, given (2) above — if not, the
   check needs a content test; if so, the finding text should say a heading is sufficient.

### PR DESCRIPTION
**Pull request by pbakaus**
Closes #436.

## Summary

- Count machine-readable `colors`, `typography`, and `components` frontmatter as valid `DESIGN.md` coverage in `doctor.mjs`.
- Preserve the existing markdown-body section behavior.
- Add a regression test for a frontmatter-only design record.

## User impact

Projects that keep canonical design tokens in `DESIGN.md` frontmatter no longer receive a contradictory `design-md-coverage` warning asking them to duplicate those values in prose sections.

## Reproduction

Before this change, a `DESIGN.md` with populated frontmatter for all three sections still emitted `design-md-coverage` naming `colors`, `typography`, and `components` as absent. The new regression test reproduces that input shape and now stays quiet.

## Validation

- `node --test tests/doctor.test.mjs`
- `bun run build`
- `bun run test`

## Generated output

Generated provider and root harness output was intentionally omitted. The source-first build completed successfully with `--skip-root-sync`, as required for feature PRs.

## AI assistance

AI assistance was used to triage and reproduce the issue, implement the focused fix, add regression coverage, run validation, and prepare this pull request.

---

> [!NOTE]
> **Low Risk**
> Narrow change to doctor staleness heuristics and tests; no auth, data, or runtime hook paths affected.
> 
> **Overview**
> Fixes **false `design-md-coverage` warnings** when canonical design tokens live in `DESIGN.md` YAML frontmatter instead of `## Colors`, `## Typography`, and `## Components` prose.
> 
> `checkDesignCoverage` in `staleness-deep.mjs` now treats a section as covered if either the parsed markdown section exists **or** `model.frontmatter[section]` passes a new `hasCoverageValue` check (non-empty strings, nested objects/arrays with real content; empty mappings, `[]`/`{}` string literals, and boolean/numeric scalars do not count).
> 
> Regression tests in `doctor.test.mjs` cover frontmatter-only records, partial empties, and edge-case scalar/collection literals.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f274ca2c013a04cdafe8498661f2155ac0d0ed88. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: skill/scripts/lib/staleness-deep.mjs [hunk 1]
```diff
@@ -126,7 +126,7 @@ export function checkDesignCoverage({ design, designPath, parseDesignMd }) {
     return [];
   }
   const missing = ['colors', 'typography', 'components']
-    .filter((section) => !model[section]);
+    .filter((section) => !model[section] && !model.frontmatter?.[section]);
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply skill/scripts/lib/staleness-deep.mjs]: Fixed in `7a0489bd`. `hasCoverageValue()` now recursively requires actual scalar content, so empty or nested-empty frontmatter mappings remain uncovered. The new regression asserts that an empty `colors:` mapping still emits `design-md-coverage` for colors while populated typography/components remain accepted. Revalidated with `node --test tests/doctor.test.mjs`, `bun run build`, and the full `bun run test` suite. AI assistance was used for this review follow-up.

File: skill/scripts/lib/staleness-deep.mjs [hunk 2]
```diff
@@ -117,6 +117,15 @@ export function checkDesignDrift({ designPath, projectRoot, threshold = 25 }) {
  * a section can be absent because it never applied, so this is reported as a
  * documentation gap for a human to judge, never as an error.
  */
+function hasCoverageValue(value) {
+  if (Array.isArray(value)) return value.some(hasCoverageValue);
+  if (value && typeof value === 'object') {
+    return Object.values(value).some(hasCoverageValue);
+  }
+  if (typeof value === 'string') return value.trim().length > 0;
+  return value !== null && value !== undefined;
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply skill/scripts/lib/staleness-deep.mjs]: Fixed in de9d5438. hasCoverageValue() now accepts only non-empty strings at scalar leaves while recursively checking populated objects and arrays, so booleans and numbers no longer count as section coverage. Added regressions for both colors: false and colors: 0; the focused doctor tests, source build, and full test suite all pass. AI assistance was used to prepare and validate this fix.

File: skill/scripts/lib/staleness-deep.mjs [hunk 3]
```diff
@@ -117,6 +117,15 @@ export function checkDesignDrift({ designPath, projectRoot, threshold = 25 }) {
  * a section can be absent because it never applied, so this is reported as a
  * documentation gap for a human to judge, never as an error.
  */
+function hasCoverageValue(value) {
+  if (Array.isArray(value)) return value.some(hasCoverageValue);
+  if (value && typeof value === 'object') {
+    return Object.values(value).some(hasCoverageValue);
+  }
+  if (typeof value === 'string') return value.trim().length > 0;
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply skill/scripts/lib/staleness-deep.mjs]: Fixed in f274ca2c. Coverage now rejects empty inline collection literals (`[]` and `{}`) while continuing to accept populated arrays. Added focused regressions for both empty collection forms and for populated arrays; `node --test tests/doctor.test.mjs`, `bun run build`, and the full `bun run test` suite all pass. AI assistance was used to prepare and validate this fix.
